server/src/controllers/attainable.ts

```diff
@@ -151,6 +161,70 @@ export async function addAttainable(req: Request, res: Response): Promise<void>
   });
 }
 
+export async function deleteAttainable(req: Request, res: Response): Promise<void> {
+  /*
+   * TODO: Check that the requester is logged in, 401 Unauthorized if not
+   * TODO: Check that the requester is authorized to add a course instance, 403
```

Comment:
This question does not only concern this PR but in general editing/deleting attainables.

Should each attainable have owner (reference to user's id) who is the only one allowed to edit/delete the attainable and its subattainables. Or should any user marked as teacher for the instance allowed to edit/delete any attainable on that instance? Or should editing be allowed for all teachers but deleting just for the attainable owner?

Reply:
For a definitive answer we'll have to ask the PO, but from what I understand, attainables probably don't need any special access rights which can't be implemented with course instance roles. I would guess that teachers in charge have rights to edit, add, and delete all attainables of the instance, and other teachers can only add points.